### PR TITLE
7591 abi encoding optimization

### LINKED ISSUE #7591: Optimize ABI API to have less allocation
Encoding and Decoding ABI allocates a lot, rewrite it to a version that will allocate a lot less (maybe streams?) and update usages.

### PR DESCRIPTION
**Pull request by ssonthal**
Fixes Closes Resolves #7591 

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Changes

- _List the changes_

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

#### Requires documentation update

- [ ] Yes
- [x] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/bench_precompiles
Comment (ssonthal) [inline src/bench_precompiles:1]: Please ignore this file. Will undo these changes once I have some confirmations from your end.

**Comment by ssonthal** [comment]
Hey @LukaszRozmej , awaiting your feedback on this.

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]
hard  to say without example usages, the one for address doesn't do much

File: src/Nethermind/Nethermind.Abi/AbiAddress.cs [hunk 1]
```diff
@@ -21,14 +24,22 @@ private AbiAddress() : base(160)
...
-                            return packed ? bytes : bytes.PadLeft(UInt256.LengthInBytes);
+                            if (packed)
+                            {
+                                nettyAbiStream.Write(input.Bytes.ToList());
+                            }
+                            else
+                            {
+                                nettyAbiStream.Write(input.Bytes.PadLeft(UInt256.LengthInBytes).ToList());
+                            }
+                            return nettyAbiStream.AsSpan().ToArray();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Abi/AbiAddress.cs:42]: you allocate both list and array, not achieving goal of removing allocations

**Comment by ssonthal** [comment]
I understand. Let me go over this again and get back.

**Comment by ssonthal** [comment]
Closing this PR. Starting the work again on a new PR.

**Pull request closed without merging**